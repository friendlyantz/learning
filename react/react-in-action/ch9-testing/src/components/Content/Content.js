import React from "react"
import { Component } from "react-dom"

const Content = (props) => {
  return (
    <div className="content">
      hello content
    </div>
  )
}

export default Content
