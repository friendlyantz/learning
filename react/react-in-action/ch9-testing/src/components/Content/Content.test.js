import React from "react"
import ShallowRenderer from "react-test-renderer/shallow"
import renderer from "react-test-renderer"
import { shallow } from "enzyme"
import Content from "./Content"

describe("<Content />", () => {
  test("should render correctly", () => {
    const mockPost = {
      content: "hello content"
    }
    const renderer = new ShallowRenderer()
    renderer.render(<Content post={mockPost}/>)
    const result = renderer.getRenderOutput()
    expect(result.type).toBe('div')
    expect(result.props.children).toEqual('hello content')
  })
})

describe("<Content />", () => {
  test("should render correctly", () => {
    const mockPost = {
      content: "I am learning to test react components"
    }
    const wrapper = shallow(<Content post={mockPost} />)
    expect(wrapper.find('div').length).toBe(1)
    expect(wrapper.find('div').text()).toBe(mockPost.content)
  })

  test("snapshot", () => {
    const mockPost = {
      content: "I am learning to test react components"
    }
    const component = renderer.create(<Content post={mockPost} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
