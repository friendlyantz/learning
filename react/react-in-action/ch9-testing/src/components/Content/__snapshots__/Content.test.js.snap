// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> snapshot 1`] = `
<div
  className="content"
>
  I am learning to test react components
</div>
`;
